### PR TITLE
[Woo POS] Add missing transitions to Cart

### LINKED ISSUE #13747: [Woo POS] Add missing transitions to product and cart views
Parent task: #13700

p91TBi-bHy-p2

#### Products & Cart
- [x] ‘Product added to cart’ fade in: fade in a row and separately fade in row content more slowly [4:15]
- [x] Consider updating "Check out" animation when Cart moves from right to left. Sometimes subviews are not animating together

### PR DESCRIPTION
**Pull request by staskus**
Closes: #13747

## Description

Adding missing transitions to cart:
- When item is added
- When item is removed
- When Cart transitions to/from empty state

I opted for faster-than-default animations for a snappier feeling.

## Steps to reproduce

1. Open POS
2. Add item
3. Confirm transition from empty state
4. Add more items
5. Confirm transition when item is added
6. Remove the last item
7. Confirm transition
8. Remove the middle item
9. Confirm other items animate into the positions
10. Clean the cart
11. Confirm the transition to empty state

## Testing information

No unit tests, UI-changes only.

## Screenshots

https://github.com/user-attachments/assets/fce6b1e6-50e5-48d7-a8a9-85c56312e422

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/221">20.2</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by staskus** [comment]
There's one Cart-Totals related animation I would like to improve. If I find a proper solution, I'll make another PR. The Cart doesn't always transition smoothly to Totals view. Not all views within Total views are moving together. An example in slow-mo:

https://github.com/user-attachments/assets/99ebfa81-18c3-4455-885d-41abd37376fa

File: WooCommerce/Classes/POS/Presentation/Card Present Payments/Reader Messages/PointOfSaleCardPresentPaymentCaptureErrorMessageViewModel.swift [hunk 1]
```diff
@@ -26,7 +26,12 @@ final class PointOfSaleCardPresentPaymentCaptureErrorMessageViewModel: Observabl
     }
 
     static func == (lhs: PointOfSaleCardPresentPaymentCaptureErrorMessageViewModel, rhs: PointOfSaleCardPresentPaymentCaptureErrorMessageViewModel) -> Bool {
-        lhs.id == rhs.id
+        lhs.id == rhs.id &&
```
Comment (staskus) [inline WooCommerce/Classes/POS/Presentation/Card Present Payments/Reader Messages/PointOfSaleCardPresentPaymentCaptureErrorMessageViewModel.swift:29]: Addressed a comment from the previous PR https://github.com/woocommerce/woocommerce-ios/pull/13741#discussion_r1732411976

File: WooCommerce/Classes/POS/Presentation/Card Present Payments/Reader Messages/PointOfSaleCardPresentPaymentCaptureErrorMessageViewModel.swift [hunk 2]
```diff
@@ -26,7 +26,12 @@ final class PointOfSaleCardPresentPaymentCaptureErrorMessageViewModel: Observabl
...
+        lhs.title == rhs.title &&
+        lhs.message == rhs.message  &&
```
Comment (iamgabrielma) [inline WooCommerce/Classes/POS/Presentation/Card Present Payments/Reader Messages/PointOfSaleCardPresentPaymentCaptureErrorMessageViewModel.swift:31]: One missing!

```suggestion
        lhs.message == rhs.message  &&
        lhs.nextStep == rhs.nextStep &&
```

**Review by iamgabrielma — APPROVED** [review]
LGTM 🚢

**Pull request merged**